pennylane/ops/functions/eigvals.py

```diff
@@ -14,20 +14,30 @@
 """
 This module contains the qml.eigvals function.
 """
+import warnings
+
 # pylint: disable=protected-access
 from functools import reduce
-import warnings
+
+import scipy
 
 import pennylane as qml
 
 
 @qml.op_transform
-def eigvals(op):
+def eigvals(op, k=1, which="SA"):
     r"""The eigenvalues of one or more operations.
 
+    For a SparseHamiltonian object, the eigenvalues are computed with the efficient ``scipy.sparse.linalg.eigsh``
+    method which returns `k` eigenvalues. For an :math:`N \times N` sparse matrix, `k` must be smaller than `N - 1`,

```

Comment:
Add a test for the else clause. (i.e a `SparseHamiltonian` is provided but  N - 1 <= k <= N)

Reply:
Added two more tests.